drivers/adc/Kconfig

```diff
@@ -77,239 +74,12 @@ config ADC_1_IRQ_PRI
 	prompt "ADC 1 interrupt priority"
 	default 2
 
-menuconfig ADC_TI_ADC108S102
-	bool "TI adc108s102 chip driver"
-	select SPI
-	select ADC_0
-	default n
-	help
-	Enable support for TI's ADC chip adc108s102 driver.
-
-if ADC_TI_ADC108S102
-
-config ADC_TI_ADC108S102_SPI_PORT_NAME
-	string "Master SPI port name"
-	default ""
-	help
-	  Master SPI port name through which adc108s102 chip is accessed.
-
-config ADC_TI_ADC108S102_SPI_CONFIGURATION
-	hex "Master SPI port configuration"
-	default 0x0
-	help
-	  Master SPI port configuration flags used to access adc108s102 chip.
-
-config ADC_TI_ADC108S102_SPI_MAX_FREQ
-	hex "Master SPI port max frequency"
-	default 0x0
-	help
-	  Master SPI port maximum frequency used to access adc108s102 chip.
-
-config ADC_TI_ADC108S102_SPI_SLAVE
-	int "SPI slave slot"
-	default 0
-	help
-	  adc108s102 chip's SPI slave number on master SPI port.
-
-endif # ADC_TI_ADC108S102
-
-menuconfig ADC_DW
-	bool "ARC Designware Driver"
-	depends on ARC
-	select ADC_0
-	default n
-	help
-	  Enable the driver implementation of the Designware ADC IP.
-
-if ADC_DW
-
-
-config ADC_DW_CALIBRATION
-	bool "Enable Calibration"
-	default y
-	help
-	  Enables ADC to run with a calibrated output at the
-	  expense of execution speed when exiting low power states.
-	  If disabled, the ADC will require the application/system-integrator
-	  to provide a calibration method.
-
-config ADC_DW_DUMMY_CONVERSION
-	bool "Enable dummy conversion"
-	default y
-	help
-	  After awaking from low power state a dummy
-	  conversion must be performed and discarded.
-	  If disabled the user will have to discard the first
-	  sample after a resume from a low power state.
-
-choice
-	prompt "Output Mode"
-	default ADC_DW_SERIAL
-	help
-	  ADC output mode: parallel or serial.
-
-config ADC_DW_SERIAL
-	bool "Serial"
-
-config ADC_DW_PARALLEL
-	bool "Parallel"
-
-endchoice
-
-choice
-	prompt "Sequence Mode"
-	default ADC_DW_SINGLESHOT
-	help
-	  ADC sequence mode - single run/repetitive
-
-config ADC_DW_SINGLESHOT
-	bool "Single Ended"
-
-config ADC_DW_REPETITIVE
-	bool "Differential"
-
-endchoice
-
-choice
-	prompt "Capture Mode"
-	default ADC_DW_RISING_EDGE
-	help
-	  ADC controller capture mode:
-	  by rising or falling edge of adc_clk
-
-config ADC_DW_RISING_EDGE
-	bool "Rising Edge"
-
-config ADC_DW_FALLING_EDGE
-	bool "Falling Edge"
-
-endchoice
-
-config ADC_DW_SAMPLE_WIDTH
-	int "Sample Width"
-	default 31
-	help
-	  Defines ADC device data sample width (resolution):
+source "drivers/adc/Kconfig.dw"
 
-	  - 0 = 6 bits resolution
+source "drivers/adc/Kconfig.mcux"
 
-	  - 1 = 8 bits resolution
+source "drivers/adc/Kconfig.qmsi"
```

Comment:
Couldn't we replace the three source lines above with:
```
source "drivers/adc/Kconfig.*"
```
this would remove future dependency on this file when new drivers are added.